src/Identity/Core/src/IdentityApiEndpointRouteBuilderExtensions.cs

```diff
@@ -42,44 +59,64 @@ public static IEndpointConventionBuilder MapIdentityApi<TUser>(this IEndpointRou
         {
             var userManager = sp.GetRequiredService<UserManager<TUser>>();
 
+            if (!userManager.SupportsUserEmail)
+            {
+                throw new NotSupportedException($"{nameof(MapIdentityApi)} requires a user store with email support.");
+            }
+
+            var userStore = sp.GetRequiredService<IUserStore<TUser>>();
+            var emailStore = (IUserEmailStore<TUser>)userStore;
+
             var user = new TUser();
-            await userManager.SetUserNameAsync(user, registration.Username);
+            await userStore.SetUserNameAsync(user, registration.Username, CancellationToken.None);
+            await emailStore.SetEmailAsync(user, registration.Email, CancellationToken.None);
```

Comment:
Should we get the `HttpContext` and pass `RequestAborted` to these?

Reply:
This copies Identity UI. I'm not sure it makes sense to cancel a database transaction just because the request is aborted. In practice with the EF stores, these all complete immediately after changing the in-memory representation. Only UserManager.CreateAsync below actually commits anything to the database and that doesn't accept a cancellation token.

There is the [AspNetUserManager](https://github.com/dotnet/aspnetcore/blob/a56e968c19be1275db6dc462310d723615b006a7/src/Identity/Core/src/AspNetUserManager.cs) that attaches the RequestAborted token to the UserManager whether it makes sense or not. I think that's at least all or nothing for user creation. We don't want a partially created user which might be technically possible for the preview7 backport but is fixed here by [`dcd5609` (#49498)](https://github.com/dotnet/aspnetcore/pull/49498/commits/dcd560997bcdac44e96a570cc43c197ff6e8daa0) 😆.